src/components/app/details/triggerView/triggerViewConfigDiff/TriggerViewConfigDiff.tsx

```diff
@@ -50,9 +54,25 @@ export default function TriggerViewConfigDiff({
     const [secretOptionCollapsed, setSecretOptionCollapsed] = useState<boolean>(false)
     const [currentData, setCurrentData] = useState<any>({}) // store codeEditorValue of current(lhs) and base(rhs) config
 
+    useEffect(() => {
+        const newSearchParams = {
+            ...searchParams,
+            config: searchParams.config?.replace('-', '/'),
+            deploy: searchParams.deploy,
+        }
+        if (canReviewConfig() && baseTemplateConfiguration && currentConfiguration) {
+            const configParamValue = searchParams.config?.replace('-', '/')
+            history.push({
+                search: new URLSearchParams(newSearchParams).toString(),
+            })
+            //handling the case when the user directly lands on the deployment history page
+            handleNavOptionSelection(null, configParamValue)
+        }
+    }, [canReviewConfig(), baseTemplateConfiguration, currentConfiguration])
```

Comment:
canReviewConfig should be some boolean value instead of method

Reply:
It is a call back function that return boolean